File: javascript/examples/arithmetic_mean.js
var averages = require('../lib/mean_averages.js');
var fs = require('fs');
var readline = require('readline');

let sampleNumber = 1;

var sampleData = readline.createInterface({
    input: fs.createReadStream('../../sample_data'),
    console: false
});

sampleData.on('line', function(line) {
    line = line.trim();
    if (line.length == 0) {
        return;
    }

    let samplePoints = line.split(",");
    let sample = [];

    samplePoints.forEach(element => {
        sample.push(parseFloat(element));
    });

    let average = averages.arithmeticMean(sample);

    printData(sampleNumber++, sample, average);
});

function printData(sampleNumber, sample, average) {
    console.log('Sample ' + sampleNumber);
    console.log('Sample Data: ' + sample.join(', '));
    console.log('Arithmetic Mean: ' + average);
    console.log();
    console.log();
}

File: dart/examples/arithmetic_mean.dart
import 'dart:core';
import 'dart:io';
import 'package:clounce_math/clounce_math.dart';
import 'package:clounce_math/statistics/averages.dart';

void main(List<String> args) {
  var sampleData = File('../../sample_data').readAsLinesSync();
  int sampleNumber = 1;
  var numRegex = RegExp(r"\d+(.\d+)?");

  sampleData.forEach((line) {
    if (!line.trim().isEmpty) {
      var samplePoints = numRegex.allMatches(line);
      var sample = samplePoints.map((point) => num.tryParse(point.group(0))).toList();
      var average = Averages.arithmeticMean(sample);

      printSample(sampleNumber++, sample, average);
    }
  });
}

void printSample(int sampleNumber, List<num> sample, num average) {
  print('Sample ${sampleNumber}');
  print('Sample Data: ' + sample.join(', '));
  print('Arithmetic Mean: ${average}');
  print('');
  print('');
}